setup.py

```diff
@@ -1,63 +1,44 @@
-from setuptools import setup
+from setuptools import setup, Extension
 import os
-
-# make sure users without cython can install our extensions
-try:
-    from Cython.Distutils.extension import Extension
-    from Cython.Distutils import build_ext as _build_ext
-    USE_CYTHON = True
-except ImportError:
-    from setuptools import Extension
-    from setuptools.command.build_ext import build_ext as _build_ext
-    USE_CYTHON = False
-
-print('using cython', USE_CYTHON)
-
-
-# make sure numpy is installed before we try to build
-# the extenion
-class build_ext(_build_ext):
-    def finalize_options(self):
-        super().finalize_options()
-        import numpy
-        self.include_dirs.append(numpy.get_include())
-
+from Cython.Build import cythonize
+import numpy as np
+
+
+kwargs = dict(
+    include_dirs=[np.get_include()],
+    define_macros=[
+        # fixes a warning when compiling
+        ("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION"),
+        # defines the oldest numpy we want to be compatible with
+        ("NPY_TARGET_VERSION", "NPY_1_21_API_VERSION"),
+    ]
+)
 
 # if we have cython, use the cython file if not the c file
```

Comment:
This comment can probably be deleted as well

Reply:
Yes, thanks, fixed